tools/toolchains/gcc.py

```diff
@@ -48,14 +48,16 @@ def __init__(self, target,  notify=None, macros=None, build_profile=None,
 
         tool_path = TOOLCHAIN_PATHS['GCC_ARM']
         # Add flags for current size setting
-        default_lib = "std"
+        default_lib = ""
         if hasattr(target, "default_lib"):
+            self.check_c_lib_supported(target, "gcc_arm")
             default_lib = target.default_lib
         elif hasattr(target, "default_build"):
             default_lib = target.default_build
```

Comment:
else raised an exception since `default_lib` would not be set

Reply:
say that "default_lib is not set"

Reply:
I do not think there is a need to raise an exception. `target.default_lib` and `target.default_build` have simply not been found that's it. `default_lib` is just a variable to get the content of either parameter if they exist, if we have not found them it's fine.

Reply:
It used to be set to "std" by default. Have we tested with a targets.json that does not specify `default_lib`?

Reply:
`"std"` was not checked for anything. It could have been `""` and it would have worked the same.`"std"` is given a meaning in this PR, therefore the default value had to be changed to something else. An empty string was fit for purpose.

Reply:
When we do renaming default_lib to c_lib and will remove default_build keyword

Reply:
What point are you making here? I am confused.

Reply:
we will remove the below code and only call self.check_c_lib_supported(target, "gcc_arm")
``` if hasattr(target, "default_lib"): 
          self.check_c_lib_supported(target, "gcc_arm") 
           default_lib = target.default_lib 
        elif hasattr(target, "default_build"): 
           default_lib = target.default_build```

Reply:
I do not see what that has to do with what we are talking about on this thread.
We are talking about the state of the code now and what should be done for it to be correct. We are not talking about how it should be in the future.

Reply:
Yes true, I am just notifying that's going to change and mentioned that I will keep exception now.

Reply:
It is not a justification to keep it because as I said the variable `default_lib` does not have to be set to anything. It only matters if has the value `"small"` otherwise we dont care about it and there is no need to raise an exception.

Reply:
I changed the code like below in case if any particular target overwrote default_build with null 
```
        # Add flags for current size setting                                    
        default_lib = ""                                                        
        if hasattr(target, "default_lib"):                                      
            self.check_c_lib_supported(target, "gcc_arm")                       
            default_lib = target.default_lib                                    
        elif hasattr(target, "default_build"):                                  
            default_lib = target.default_build                                  
                                                                                
        if not default_lib:                                                     
            raise NotSupportedException("default_lib is not set")   
```

Reply:
Hugues is right, not exception required. You could leave the code as it was originally.